src/partitioned_data/M_partitioned_data.jl

```diff
@@ -303,4 +308,42 @@ function evaluate_grad_part_data!(part_data::PartitionedData)
   return part_data
 end
 
+"""
+    hv = part_hprod(part_data::PartitionedData, x::AbstractVector, v::AbstractVector)
+
+Return `hv::Vector` the product between the partitioned hessian ∇²f(`x`) and the vector `v`.
+"""
+function part_hprod(part_data::PartitionedData, x::AbstractVector, v::AbstractVector)
```

Comment:
I think we might as well use multiple dispatch since we have it. If I read
```julia
hprod(part_data::PartitionedData, x::AbstractVector, v::AbstractVector)
```
I know immediately that we're talking about `hprod` applied to `PartitionedData` objects. But if it's called `part_hprod`, I'm not so sure.